lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -774,7 +774,16 @@ handle_heredocs(T, Line, Column, H, Scope, Tokens) ->
 handle_strings(T, Line, Column, H, Scope, Tokens) ->
   case elixir_interpolation:extract(Line, Column, Scope, true, T, H) of
     {error, Reason} ->
-      interpolation_error(Reason, [H | T], Scope, Tokens, " (for string starting at line ~B)", [Line]);
+      {error, {Meta, Message, Token}, NewRest, NewScope, NewTokens} = interpolation_error(Reason, [H | T], Scope, Tokens, " (for string starting at line ~B)", [Line]),
+      NewMeta = case lists:keyfind(error_type, 1, Meta) of
```

Comment:
Shouldn't this matching be handled inside `interpolation_error` instead? 🤔  This way we don't need to make changes to bidi (i.e. all other errors pass through).

Reply:
I also think this code is called for `'foo'`. So you need to use the variable `H` to detect which kind of delimiter we are expecting.

Reply:
> Shouldn't this matching be handled inside interpolation_error instead?
From my tests and understanding, we get the bidi error from elixir_interpolation:extract, and then interpolation_error just passes it through

I've updated bidi to add its error type so we can differentiate and not override it, because the way it currently is we receive the same error structure whether it's a bidi error or an unclosed string one

> I also think this code is called for 'foo'

Great point, I'll do some tests with single-quote strings, I believe they should have this same behavior if unclosed, right?

Reply:
> I've updated bidi to add its error type so we can differentiate and not override it, because the way it currently is we receive the same error structure whether it's a bidi error or an unclosed string one

I think you only receive the same structure because you are intercepting the error. Instead of intercepting it, change where the error is raised to already include all metadata.